### ISSUE TITLE
How Rules become Active

### ISSUE THREAD

**Issue by nitmws** [w3c/poe#221]
The IM of 21 August defines in 2.5 Rule Class:

> For a Rule to become effective, all of it's Constraints must be satisfied and any applicable relationship to a Duty must be fulfilled.

Note: today we agreed the term "effective" is replaced by "active".

Checking if a Constraint is satisfied or not its leftOperand must be matched against an - in most cases - stable value, e.g. action must not be taken before 1 January 2018, asset can only be presented on mobile devices.
That means: the rightOperand-date must be matched against the date-time of the planned action and the planned distribution channel of the asset must be matched against the rightOperand "mobile device"

This raises the question: a Rule may have also an assigner and an assignee and should have a target - should they be matched too against the value of a planned action?
In other words: is a Rule with assignee = "http://examples.com/company/4711" active if it is evaluated by assignee = "http://examples.com/company/9912"?
I guess ODRL users will expect it is not-active, as this is the same situation as with not satisfied Constraint(s).

**Comment by simonstey** [comment]
> In other words: is a Rule with assignee = "http://examples.com/company/4711" active if it is evaluated by assignee = "http://examples.com/company/9912"?

no

**Comment by simonstey** [comment]
(or active but not valid)

**Comment by nitmws** [comment]
Coming back to the current definition:

> For a Rule to become effective, all of it's Constraints must be satisfied and any applicable relationship to a Duty must be fulfilled.

Evaluating if a Constraint is satisfied or not is covered by the IM.

But what what about "any applicable relationship to a Duty must be fulfilled."
* what makes a Duty "applicable" and what not?  Is this equivalent to: the Duty is Active or Not-Active?
* a ambiguous requirement is "Duty must be fulfilled": example: a duty requires to pay 100 EUR. But what if the asset of the Permission is not an asset the assignee of the Policy wants to use? The receiver will definitely not pay. In this case the duty is not fulfilled and therefore the Permission including it Not-Active.
* But that result builds on a side effect: the duty is not not-fulfilled because the assignee does not want to or has not paid yet; it is not fulfilled as the - let's call it - context of the evaluation of the Rule does not match the Rule.
* I think ODRL has to include such a context into any realistic evaluation of "Active - Not-Active". If a Rule includes a constraint "Use only after 1 January 2018" the context must deliver the current date.
* I think this is a strong reason why checking the assignee and the target property values of a Rule against a context defining both should be part of an "Active or Not-Active" evaluation.

**Comment by nitmws** [comment]
I've created a draft of guidelines for evaluating if a Permission, Prohibition or Duty is Active or Not-Active:
https://docs.google.com/spreadsheets/d/1vej4EwEZjUM2yYJwBggJxHlSTCinyuMPCulKIeHNTA0/edit?usp=sharing
... in the upper section of the table.

The key specification in the IM for these guidelines is the definition mentioned above.
These guidelines build on "applicable" Duty = Active Duty in the duty(ies), remedy(-ies) and Duty's consequence(s) columns.
If a Duty is fulfilled or not is by my understanding an inconsistent requirement:
* For the Not-/Active evaluation of a Rule the action of the Rule is not relevant
* For the Not-/Active evaluation of a Duty - which is a sub-class of Rule - the action is suddenly relevant.

I'm not convinced that a Not-/Active status of a Duty is relevant for the Not-/Active evaluation of the Rule including this Duty, it has primarily an impact on the evaluation of the result of a Rule, see #223, so nothing gets lost. (In #223 I mention that the split between Not-/Active evaluation and result evaluation is a free decision, ODRL could also combine both in a single evaluation.)
Therefore I have greyed out the duty(-ies) and remedy(-ies) column of the guideline table.

**Comment by riannella** [comment]
> what makes a Duty "applicable" and what not? 

My mistake. It is now "...and all relationships..."

**Comment by riannella** [comment]
> a ambiguous requirement is "Duty must be fulfilled": example: a duty requires to pay 100 EUR. But what if the asset of the Permission is not an asset the assignee of the Policy wants to use? The receiver will definitely not pay. In this case the duty is not fulfilled and therefore the Permission including it Not-Active.

I can't get my head around this ;-)

If the assignee does not want to use the asset, then why would there even be a policy?

Another way to look at this: Michael has a Permission to use Asset X if he pays Renato $50.
Michael does not want to use Asset X.
End of transaction.

I think the party/asset's are assumed to be the ones expressed in the Policy, otherwise there is no point in doing any form of evaluation.

**Comment by nitmws** [comment]
Re my concerns regarding "Duty must be fulfilled" to make the Rule including it Active

A/ What is the role of Not-/Active in the evaluation of a Rule
(Note: Active = in effect = Active)
* The IM of 21 August outlines in 1.3 Terminology that "in effect" is needed by an ODRL Evaluator, but chapter 3 is currently "under construction" ...
* ... the shown "old" version told "Where no rules are in effect, the policy is effectively silent on what actions may or may not be performed." ...
* ... in other words: if a rule is Not-Active it is (= has to be) ignored by the Evaluator. As I recollect discussions this is shared by the WG persons.

B/ Why does it make sense to set the state Not-/Active
* The evaluation of a Rule in compliance with ODRL could be done without having such a state, so a practical use is linked to setting this state
* This practical use appears to be: the receiver applies a specific use context to a received Policy, the context defines the id of the assignee, the id of an (collection of) asset(s), an action and more like when and where and in which way it should be used ...
* ... for this case it is of practical help to sort out the Rules of a Policy which match this context. 
Currently the evaluation of the Not-/Active state does not include assignee, target and action but it goes  over the Rules and checks by defined criteria for this state. 
* This helps to exclude Rules which should not be further evaluated for formal reasons related to the use context: if the asset should be used now time constraints should be evaluated, if the asset should be used in France spatial constraints should be evaluated, if the asset should be used with a mobile app delivery channel constraints should be evaluated.
* A use case of this Not-/Active state is e.g. to list all Active Permissions and Prohibitions which match the assignee and the asset id, then the assignee could pick Permissions of interest and check not to violate Prohibitions. A real case: the asset id is the id of a feed of photos of a photo agency. The assignee is a publisher and its editorial system should show all photos which may be used _today_ for the _mobile app_ of the publisher. From the shown photos an editor will pick 1 and use it with a news story. (Btw: such a case is done by the [British Guardian](https://www.theguardian.com) already now.)

C/ Why not evaluating "Duty must be fulfilled" for the Not-/Active state
* In such a use case requiring "Duty must be fulfilled" would result in a disaster.
Checking if the constraints are Satisfied may result in multiple Permission as potentially Active. Does the assignee has to **execute all duties**, including paying fees, to make these Permissions Active? This is what "Duty must be fulfilled"requires, but the assignee will not do that.
* A realistic workflow is: only the constraints are checked, all Permissions stated as Active are listed, the assignee picks one out of t he many and to get the Permission result Has-Permission the duty has to be executed, including paying a fee.

Note: yes there is another approach to the Rules in a Policy: the assignee sets a context for using a single asset only and the Policy has only 1 Permission for this context. In this special case there is no practical difference between having to fulfil the duty to make the Permission Active and getting an Active Permission without a fulfilled duty and having to fulfil it when taking the step towards Has-Permission. But getting only a single Permission out of a Policy is a special and not the generic use case.

**Comment by nitmws** [comment]
@riannella what do you mean by:

> I think the party/asset's are assumed to be the ones expressed in the Policy, 

Does that mean the are expressed at the Policy-level making it a Compact Policy?
therefore: only a single assignee and a single asset ID exists in the whole Policy?

This WG should not assume Policies are strictly tailored to the use of a single receiver/assignee.
(_IPTC had to deal with this use case: a big publishing company runs 3 news publications and has only 1 IT system - and the company has different contracts for the publications with a news agency (= the assignee)._ )

**Comment by nitmws** [comment]
My conclusion as of 28 August: what "active"/"in-effect" means gets clearer when merging all 20 occurrences of this term in the IM and in other documents - how foggy the use of the term is was written down in #225
